lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -53,12 +53,16 @@ def available_version
     version_field.try(:fetch_path, 'field', 'default_value')
   end
 
-  def self.import_git_repo(domain_name, git_repo, ref = DEFAULT_BRANCH, ref_type = BRANCH)
-    MiqAeDomain.find_by(:name => domain_name).try(:destroy)
-    MiqAeImport.new(domain_name, import_options(git_repo, ref, ref_type)).import
-    domain = MiqAeDomain.find_by(:name => domain_name)
-    raise MiqAeException::DomainNotFound, "Domain #{domain_name} not found after import" unless domain
-    domain.update_git_info(git_repo, ref, ref_type)
+  def self.import_git_repo(options)
+    git_repo = GitRepository.find(options['git_repository_id'])
+    raise "Git repository with id #{options['git_repository_id']} not found" unless git_repo
+
+    MiqAeDomain.find_by(:name => options['domain']).try(:destroy) if options['domain']
+    import_options(git_repo, options)
+    domain = Array.wrap(MiqAeImport.new(options['domain'] || '*', options).import).first
```

Comment:
Do we need the `Array.wrap` if we're just calling `.first` on it? Seems like this could just be `MiqAeImport.new(options['domain'] || '*', options).import`